roles/ssl-cert/tasks/main.yml

```diff
@@ -0,0 +1,51 @@
+---
+# install required packages
+- name: Install required packages
+  yum: name="{{ item }}" state=latest
+  with_items:
+  - bind-utils
+  - hostname
+  - openssl
+
+- name: Prepare tmp directory.
+  tempfile:
+    prefix: "ssl-certs."
+    state: directory
+  register: ssl_cert_tmp_dir
+
+- name: Prepare helper script ssl_certs.sh
+  copy: src=ssl_cert.sh dest="/opt/ssl_cert.sh" mode="u=rwx,g=rx,o=rx"
+
+- name: Generate certs
+  command: "/opt/ssl_cert.sh {{ ssl_cert_tmp_dir.path }} {{ ssl_cert_name }}"
+
+- name: Copy generated cert to desired location
+  copy:
+    src: "{{ ssl_cert_tmp_dir.path }}/{{ ssl_cert_name }}.crt"
+    dest: "{{ ssl_certs_dir }}/{{ ssl_cert_name }}.crt"
+    remote_src: yes
+    mode: 0644
+    owner: "{{ ssl_owner}}"
+    group: "{{ ssl_group }}"
+
+- name: Copy generated key to desired location
+  copy:
+    src: "{{ ssl_cert_tmp_dir.path }}/{{ ssl_cert_name }}.key"
+    dest: "{{ ssl_keys_dir }}/{{ ssl_cert_name }}.key"
+    remote_src: yes
+    mode: 0600
```

Comment:
It may turn out that I may need to tweak this as well, but let's wait for verification from my side ...

Reply:
Ok, could you allow this to be configured as well? It turns out I would need to set it as `0644` anyway ...

Reply:
@mbukatov  could you try it (with fc0201c6fcf22551d147a92e443436f2fa9c8bda) ... hopefully it will work :)